support/ray_cluster_client.go

```diff
@@ -45,20 +46,47 @@ type RayJobLogsResponse struct {
 	Logs string `json:"logs"`
 }
 
+type RayClusterClientConfig struct {
+	Address             string
+	Client              *http.Client
+	SkipTlsVerification bool
+}
+
 var _ RayClusterClient = (*rayClusterClient)(nil)
 
 type rayClusterClient struct {
-	endpoint url.URL
+	endpoint    url.URL
+	httpClient  *http.Client
+	bearerToken string
 }
 
 type RayClusterClient interface {
 	CreateJob(job *RayJobSetup) (*RayJobResponse, error)
 	GetJobDetails(jobID string) (*RayJobDetailsResponse, error)
 	GetJobLogs(jobID string) (string, error)
+	GetJobs() ([]map[string]interface{}, error)
 }
 
-func NewRayClusterClient(dashboardEndpoint url.URL) RayClusterClient {
-	return &rayClusterClient{endpoint: dashboardEndpoint}
+var rayClusterApiClient RayClusterClient
```

Comment:
Is it really needed? I'd rather avoid having a global variable that can introduce some side-effects. I don't think that an issue to create a new client each time the `NewRayClusterClient` function is called.